### DISCUSSION TITLE
Onedrive client for linux cannot upload files

### DISCUSSION THREAD

**Discussion by gtmarewo** [abraunegg/onedrive#2970] [Q&A]
Dear all,

Ever since I installed Ubuntu 24.04 from scratch (due to upgrage failure) I can no longer upload files to onedrive but I can download.

onedrive --version
onedrive v2.5.2-1+np1+1.1

onedrive --sync --download-only

downloads new files successfully

onedrive --sync --upload-only

returns the following error message
```
OneDrive generated an error when creating this path: ./Work/Teaching/moderation/wits

ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 404 (Not Found)
  Error Reason:     Item not found
  Error Code:       itemNotFound
  Error Timestamp:  2024-11-14T09:41:11
  API Request ID:   2b3932c5-823a-499e-ae44-2344508cadc8

OneDrive generated an error when creating this path: ./Work/Teaching/moderation/wits

ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 404 (Not Found)
  Error Reason:     Item not found
  Error Code:       itemNotFound
  Error Timestamp:  2024-11-14T09:41:12
  API Request ID:   03ce961b-fd3a-432e-adb8-1f60ff33016c

Attempting to perform a database vacuum to optimise database
Database vacuum is complete
onedrive.OneDriveException@src/onedrive.d(1398): HTTP request returned status code 404 (Not Found)
{
    "error": {
        "code": "itemNotFound",
        "message": "The resource could not be found."
    }
}
----------------
??:? [0x5b57af21dd95]
??:? [0x5b57af21a60a]
??:? [0x5b57af20568f]
??:? [0x5b57af202b35]
??:? [0x5b57af200a68]
??:? [0x5b57af200433]
??:? [0x5b57af28c571]
??:? [0x5b57af289398]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x733d95da5dcc]
??:? _d_run_main2 [0x733d95da5be6]
??:? _d_run_main [0x733d95da5a3c]
??:? [0x5b57af2b5eb1]
??:? [0x733d95a2a1c9]
??:? __libc_start_main [0x733d95a2a28a]
??:? [0x5b57af1a0254]
```
Could you please assist me with resolving this issue.

I hope to hear from you soon.

Regards,

Gerald.

**Comment by abraunegg** [comment]
@gtmarewo 
This might be related to a bug that was fixed post v2.5.2.

Please build your client from 'master' amd retest.

If you still have an issue, please open a bug report and fill in all details.

**Reply by gtmarewo** [reply]
Thank you @abraunegg  for the prompt reply.

Let me give that a go then. I was using a prebuilt package (.deb).

I will come back to update you on my progress.

Regards,

Gerald

**Reply by gtmarewo** [reply]
@abraunegg 

onedrive --display-config
warns about version 8.5.0 of curl
even if

sudo apt remove curl
and installed version 8.11.0 of curl

How do I force onedrive to use the correct version of curl?

Bye for now.

Gerald.

**Reply by abraunegg** [reply]
@gtmarewo

> How do I force onedrive to use the correct version of curl?

You need to install it to the same locations as where the Ubuntu package installs it to. If the application is still complaining about curl 8.5.0 this means you have not uninstalled/updated this buggy version of curl correctly, and not updated it correctly.

To do this correctly you need to seek support from Ubuntu on upgrading curl - your distribution choice (not mine). 

#### Tips:
1. Firstly do not use Ubuntu snaps - they are junk. If you have upgraded curl using a snap .. do not do this - you are not upgrading curl.
2. Use a better distribution such as Debian 12 that provides Debian 12 Backports where you can install an updated version of curl from without issue.

**Reply by gtmarewo** [reply]
@abraunegg 

Thank you for the response.

After fiddling around I would like to report a success.

The command

systemctl --user status onedrive.service

returned

Nov 15 07:53:26 hermes onedrive[3919]: Uploading new file: Work/Teaching/moderation/UL/docs/Forms_to_be_filled.pdf ... done
Nov 15 07:53:27 hermes onedrive[3919]: The operating system sent a deletion notification. Trying to delete the item as requested
Nov 15 07:53:28 hermes onedrive[3919]: The operating system sent a deletion notification. Trying to delete the item as requested
Nov 15 07:53:28 hermes onedrive[3919]: The operating system sent a deletion notification. Trying to delete the item as requested
Nov 15 07:53:28 hermes onedrive[3919]: The operating system sent a deletion notification. Trying to delete the item as requested
Nov 15 07:53:28 hermes onedrive[3919]: Attempting to contact Microsoft OneDrive Login Service
Nov 15 07:53:28 hermes onedrive[3919]: Successfully reached Microsoft OneDrive Login Service
Nov 15 07:53:28 hermes onedrive[3919]: Starting a sync with Microsoft OneDrive
Nov 15 07:53:29 hermes onedrive[3919]: Syncing changes from Microsoft OneDrive ...
Nov 15 07:53:37 hermes onedrive[3919]: Sync with Microsoft OneDrive is complete

Here is what I did to resolve the issue.

1. I re-installed curl using: sudo apt install curl
2. I uninstalled curl using: sudo apt purge curl (I think that remove is insufficient as it may leave some configuration files lingering)
3. I installed curl (latest version) from source using: ./configure --with-ssl; make; sudo make install
4. I Installed onedrive from source using the instructions provided in the official documentation

I am seriously **considering** giving Debian testing a go. I have only used Debian stable in the past.

I really appreciate your assistance.

PS: I don't use snaps.

Regards,

Gerald.

**Comment by abraunegg** [comment]
@gtmarewo 
Thanks for the update - so confirming:

* No further application issue when built from source
* No further operational issue when using an up-to-date curl version

**Reply by gtmarewo** [reply]
@abraunegg 

You are welcome.

onedrive client for Linux is a job well done. Kudos to you.

Here is further evidence that all is well on my side now.

onedrive --user status onedrive.service 

returns

Nov 15 10:04:52 hermes onedrive[4118]: Uploading new file: Work/Teaching/moderation/spu/241115/2024 Post-graduate rates _002_.pdf ... done
Nov 15 10:04:52 hermes onedrive[4118]: Uploading new file: Work/Teaching/moderation/spu/241115/2024 SPU Claim form for external moderation _002_.docx ... done
Nov 15 10:04:53 hermes onedrive[4118]: Uploading new file: Work/Teaching/moderation/spu/241115/2024 SPU moderators report for answer scripts.pdf ... done
Nov 15 10:04:53 hermes onedrive[4118]: Uploading new file: Work/Teaching/moderation/spu/241115/00206BF92294241030141212.pdf ... done
Nov 15 10:04:53 hermes onedrive[4118]: Uploading new file: Work/Teaching/moderation/spu/241115/Machele NAPM842.pdf ... done
Nov 15 10:09:30 hermes onedrive[4118]: Attempting to contact Microsoft OneDrive Login Service
Nov 15 10:09:31 hermes onedrive[4118]: Successfully reached Microsoft OneDrive Login Service
Nov 15 10:09:31 hermes onedrive[4118]: Starting a sync with Microsoft OneDrive
Nov 15 10:09:31 hermes onedrive[4118]: Syncing changes from Microsoft OneDrive ...
Nov 15 10:09:39 hermes onedrive[4118]: Sync with Microsoft OneDrive is complete

Cheers,

Gerald.

**Reply by gtmarewo** [reply]
It looks like I prematurely closed the discussion. Sorry about that.

Cheers,

Gerald.

**Reply by abraunegg** [reply]
I like to keep them open, you never know someone might have the same issue ...

**Reply by gtmarewo** [reply]
Noted @abraunegg . Thanks.
